### PR TITLE
feat(wallet): add post for wallet with PM

### PR DESCRIPTION
**Pull request by infantesimone**
### List of changes

- rename api and product in `payment-wallet`
- add post wallet with Payment Manager
- add GET payment methods in payment-wallet api for app IO
- add feature flag `payment_wallet_with_pm_enabled` to enable wallet with Payment Manager
- add `paymentMethodId` as input of `post wallet`

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/wallet-app/api/payment-wallet/v1/_post_wallets_with_pm_policy.xml.tpl [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      </choose>
+      <set-variable name="pmSession" value="@(((IResponse)context.Variables["pm-session-body"]).Body.As<JObject>())" />
+      <return-response>
+        <set-status code="201" />
+        <set-header name="Content-Type" exists-action="override">
+          <value>application/json</value>
+        </set-header>
+        <set-body>@{
+            return new JObject(
+                          new JProperty("redirectUrl", $"https://dev.payment-wallet.pagopa.it/onboarding#sessionToken={((string)((JObject) context.Variables["pmSession"])["data"]["sessionToken"])}")
```
Comment (pietro-tota) [inline src/domains/wallet-app/api/payment-wallet/v1/_post_wallets_with_pm_policy.xml.tpl:24]: Maybe the redirect url should be expressed as a variable to be interpolated for each environment?

File: src/domains/wallet-app/api/webview-payment-wallet/v1/_base_policy.xml.tpl [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          <set-method>GET</set-method>
+      </send-request>
+       <choose>
+        <when condition="@(((IResponse)context.Variables["pm-session-body"]).StatusCode != 200)">
+          <return-response>
+            <set-status code="401" />
+          </return-response>
+        </when>
+      </choose>
+      <set-backend-service base-url="https://weudev.ecommerce.internal.dev.platform.pagopa.it/pagopa-ecommerce-payment-methods-service" />
```
Comment (pietro-tota) [inline src/domains/wallet-app/api/webview-payment-wallet/v1/_base_policy.xml.tpl:16]: Same as above

File: src/domains/wallet-app/api/payment-wallet/v1/_post_wallets_with_pm_policy.xml.tpl (hunk 1 above)
Comment (infantesimone) [reply src/domains/wallet-app/api/payment-wallet/v1/_post_wallets_with_pm_policy.xml.tpl:24]: yep :-)

**Review by pietro-tota — APPROVED** [review]
🚀

File: src/domains/wallet-app/api/payment-wallet/v1/_openapi.json.tpl [hunk 3]
```diff
@@ -22,6 +30,86 @@
...
+          {
+            "bearerAuth": []
+          }
+        ],
+        "summary": "Retrieve all Payment Methods",
+        "parameters": [
+          {
+            "name": "amount",
+            "in": "query",
+            "description": "Payment Amount",
```
Comment (pietro-tota) [inline src/domains/wallet-app/api/payment-wallet/v1/_openapi.json.tpl:49]:
```suggestion
            "description": "Payment Amount expressed in eurocents",
```
🚲

**Pull request merged**